monitoring/src/test/scala/akkaviz/events/ThroughputMeasurementFlowTest.scala

```diff
@@ -0,0 +1,69 @@
+package akkaviz.events
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.pattern
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.testkit.{TestActorRef, TestKit}
+import akkaviz.events.types.{BackendEvent, ReceivedWithId, ThroughputMeasurement}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{Matchers, WordSpecLike}
+
+import scala.concurrent.Future
+
+class ThroughputMeasurementFlowTest extends TestKit(ActorSystem("FlowTestSystem"))
+    with WordSpecLike with Matchers with ScalaFutures {
+
+  import scala.concurrent.duration._
+
+  implicit val materializer = ActorMaterializer()(system)
+
+  val firstRef = TestActorRef[SomeActor](new SomeActor, "first")
+  val secondRef = TestActorRef[SomeActor](new SomeActor, "second")
+
+  override implicit val patienceConfig = PatienceConfig(timeout = 5.seconds)
+
+  "ThroughputMeasurementFlow" should {
+
+    "not emit any measurements if there are no Received events" in {
+      val src = Source.empty[BackendEvent]
+      val sink: Sink[BackendEvent, Future[List[BackendEvent]]] = Sink.fold(List.empty[BackendEvent])((list, ev) => ev :: list)
+
+      val materialized = ThroughputMeasurementFlow(1.second).runWith(src, sink)._2
+
+      whenReady(materialized) { r =>
+        r should be('empty)
+      }
+    }
+
+    "emit proper measured value for one message" in {
+      val src = Source.single(ReceivedWithId(1, ActorRef.noSender, firstRef, "sup", true))
+      val mat = src.via(ThroughputMeasurementFlow(1.second))
+        .toMat(Sink.head[ThroughputMeasurement])(Keep.right).run()
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2355072435/issues/source?bid=3121282&fileBranchId=3217410#l42)